curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -317,6 +317,7 @@ public String getActualPath()
     public void setData(byte[] data) throws Exception
     {
         data = Preconditions.checkNotNull(data, "data cannot be null");
+        Preconditions.checkState(nodePath.get() != null, "initial create has not been processed. Call waitForInitialCreate() to ensure.");
         this.data.set(Arrays.copyOf(data, data.length));
```

Comment:
The current behavior is undefined. I can't imagine anyone is dependent on an NPE. Right?